internal/webhook/external/github_test.go

```diff
@@ -674,22 +761,100 @@ func sign(t *testing.T, s string, b []byte) string {
 	)
 }
 
-func newBody() *bytes.Buffer {
-	return bytes.NewBuffer([]byte(`
+func newGitHubEventBody(eventType string) *bytes.Buffer {
+	switch eventType {
+	case "push", "ping", "deployment":
+		return bytes.NewBuffer([]byte(`
 {
 	"ref": "refs/heads/main",
 	"before": "1fe030abc48d0d0ee7b3d650d6e9449775990318",
 	"after": "f12cd167152d80c0a2e28cb45e827c6311bba910",
 	"repository": {
-	  "html_url": "https://github.com/username/repo"
+		"html_url": "https://github.com/username/repo"
 	},
 	"pusher": {
-	  "name": "username",
-	  "email": "email@inbox.com"
+		"name": "username",
+		"email": "email@inbox.com"
 	},
 	"head_commit": {
-	  "id": "f12cd167152d80c0a2e28cb45e827c6311bba910"
+		"id": "f12cd167152d80c0a2e28cb45e827c6311bba910"
 	}
-  }	
+}
 `))
+	case "package":
+		return bytes.NewBuffer([]byte(`
+{
```

Comment:
The sample event payload has a lot of unnecessary fields. Since this is just for testing, reduce it down to only the field you need. You can do the same for the push event too. I originally included some fields that I thought we would need but then no longer needed.

Also, I think if we use `bytes.NewBufferString` we can get rid of the type-cast.

Reply:
I've only put the required fields from the [docs itself](https://docs.github.com/en/webhooks/webhook-events-and-payloads#package). I think there are few fields that are still missing.

Reply:
I've removed the extra fields. I thought anyone reading the payload would get to know that these are the required fields that a payload should have. However, they can refer to the docs for it.  Thanks!